ZuluContent/Spells/Fourth/ArchCure.cs

```diff
@@ -1,166 +1,62 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Scripts.Zulu.Engines.Classes;
+using Server.Network;
 using Server.Targeting;
+using ZuluContent.Zulu.Engines.Magic;
 
 namespace Server.Spells.Fourth
 {
-    public class ArchCureSpell : MagerySpell
+    public class ArchCureSpell : MagerySpell, ITargetableAsyncSpell<IPoint3D>
     {
         public ArchCureSpell(Mobile caster, Item spellItem) : base(caster, spellItem)
         {
         }
 
-
-        // Arch cure is now 1/4th of a second faster
-        public override TimeSpan CastDelayBase
-        {
-            get { return base.CastDelayBase - TimeSpan.FromSeconds(0.25); }
-        }
-
-
-        public override void OnCast()
+        public async Task OnTargetAsync(ITargetResponse<IPoint3D> response)
         {
-            Caster.Target = new InternalTarget(this);
-        }
+            if (!response.HasValue)
+                return;
 
-        public void Target(IPoint3D p)
-        {
-            if (!Caster.CanSee(p))
-            {
-                Caster.SendLocalizedMessage(500237); // Target can not be seen.
-            }
-            else if (CheckSequence())
-            {
-                SpellHelper.Turn(Caster, p);
-
-                SpellHelper.GetSurfaceTop(ref p);
-
-                var targets = new List<Mobile>();
-
-                var map = Caster.Map;
-                var directTarget = p as Mobile;
-
-                if (map != null)
-                {
-                    var feluccaRules = map.Rules == MapRules.FeluccaRules;
+            var target = SpellHelper.GetSurfaceTop(response.Target);
 
-                    // You can target any living mobile directly, beneficial checks apply
-                    if (directTarget != null && Caster.CanBeBeneficial(directTarget, false))
-                        targets.Add(directTarget);
+            SpellHelper.Turn(Caster, target);
 
-                    IPooledEnumerable eable = map.GetMobilesInRange(new Point3D(p), 2);
+            var targets = new List<Mobile>();
+            var map = Caster.Map;
+            
+            IPooledEnumerable eable = map.GetMobilesInRange(target, 2);
```

Comment:
Perhaps we could make a helper function for this, as I imagine we would reuse this logic every time there's an AoE spell